defuse/src/tokens/nep141/withdraw.rs

```diff
@@ -0,0 +1,112 @@
+use defuse_contracts::{
+    defuse::{
+        tokens::{
+            nep141::{FungibleTokenWithdrawResolver, FungibleTokenWithdrawer},
+            TokenId,
+        },
+        Result,
+    },
+    utils::cache::{CURRENT_ACCOUNT_ID, PREDECESSOR_ACCOUNT_ID},
+};
+use near_contract_standards::fungible_token::core::ext_ft_core;
+use near_sdk::{
+    assert_one_yocto, env, json_types::U128, near, require, serde_json, AccountId, NearToken,
+    PromiseOrValue, PromiseResult,
+};
+
+use crate::{DefuseImpl, DefuseImplExt};
+
+#[near]
+impl FungibleTokenWithdrawer for DefuseImpl {
+    #[payable]
+    fn ft_withdraw(
+        &mut self,
+        token: AccountId,
+        receiver_id: AccountId,
+        amount: U128,
+        memo: Option<String>,
+        msg: Option<String>,
+    ) -> PromiseOrValue<U128> {
+        assert_one_yocto();
+        self.internal_ft_withdraw(
+            PREDECESSOR_ACCOUNT_ID.clone(),
+            receiver_id,
+            token,
+            amount,
+            memo,
+            msg,
+        )
+        .unwrap()
+    }
+}
+
+impl DefuseImpl {
+    fn internal_ft_withdraw(
+        &mut self,
+        sender_id: AccountId,
+        receiver_id: AccountId,
+        token: AccountId,
+        amount: U128,
+        memo: Option<String>,
+        msg: Option<String>,
+    ) -> Result<PromiseOrValue<U128>> {
+        require!(amount.0 > 0, "zero amount");
+
+        self.internal_withdraw(&sender_id, [(TokenId::Nep141(token.clone()), amount.0)])?;
+
+        let ext =
+            ext_ft_core::ext(token.clone()).with_attached_deposit(NearToken::from_yoctonear(1));
+        let is_call = msg.is_some();
+        Ok(if let Some(msg) = msg {
+            ext.ft_transfer_call(receiver_id, amount, memo, msg)
```

Comment:
With static gas?

Reply:
`.with_static_gas(...)` is only used for callback below, so that the rest will be credited to `ft_transfer_call(...)`. This helps to make the caller responsible for making the decision on how much gas to pass to our function, while we do not impose any limits on calls to `ft_transfer_call(...)`. For example, a recipient can be `swap_intent` which executes heavy `action` on `ft_on_transfer(...)`, so we won't end up with reverting this tx in any case regardless of how much gas the caller passes to us.